src/firebase/users.js

```diff
@@ -88,3 +89,42 @@ export const getUserAppliedCalls = async (uid) => {
     console.log(error);
   }
 };
+
+export const getUserAppliedCallsData = async (appliedCalls) => {
+  return await Promise.all(
+    appliedCalls?.map(async (call) => {
+      const callData = await getCall(call.id);
+      const mergedCalls = { ...callData, ...call };
+      return mergedCalls;
+    })
+  );
+};
+
+export const checkUserAppliedCallDates = async (applicantID, proposedCall) => {
+  try {
+    //We do not need to consider rejected applications
+    const userAppliedCallsRef = (await getUserAppliedCalls(applicantID)).filter(
+      function (call) {
+        return call.status !== "rejected";
+      }
+    );
+
+    //Overlapping time intervals are calculated by the following clause
+    //(start1 < end2 && start1 > start2) || (start2 < end1 && start2 > start1);
+    const userBlockingAppliedCalls = (
+      await getUserAppliedCallsData(userAppliedCallsRef)
+    ).filter(function (call) {
+      return (
+        (call.date.startDate <= proposedCall.date.endDate &&
+          call.date.startDate >= proposedCall.date.startDate) ||
+        (proposedCall.date.startDate <= call.date.endDate &&
+          proposedCall.date.startDate >= call.date.startDate)
+      );
+    });
+    console.log(userBlockingAppliedCalls);
```

Comment:
Little console.log here if you have the chance can you remove it pls?

Reply:
done